### PR TITLE
Use the url spec to parse ice server urls (Take 2).

### LINKED ISSUE #2660: Unclear how URLs are processed
Chrome throws an exception for
```js
new RTCPeerConnection({ iceServers: [{ urls:["stun:turn.example.org/test"] }] });
```
whereas Firefox and Safari do not.

Where does https://w3c.github.io/webrtc-pc/#dom-peerconnection process these?

Some processing appears to happen in https://w3c.github.io/webrtc-pc/#set-the-configuration but that does not seem to be invoked from the constructor. (That also suggests using a URL parser that browsers do not have, which seems somewhat dubious.)

### PR DESCRIPTION
**Pull request by jan-ivar**
Fixes #2660. @annevk PTAL.

***
<a href="https://pr-preview.s3.amazonaws.com/jan-ivar/webrtc-pc/pull/2853.html" title="Last updated on Apr 4, 2023, 2:11 PM UTC (1a5b430)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webrtc-pc/2853/fb2be18...jan-ivar:1a5b430.html" title="Last updated on Apr 4, 2023, 2:11 PM UTC (1a5b430)">Diff</a>

### CODE REVIEW

File: webrtc.html [hunk 1]
```diff
@@ -3073,44 +3073,54 @@ <h4>
...
-                          "scheme">turn</code> or <code class=
-                          "scheme">turns</code>, and
-                          <var>server</var>.{{RTCIceServer/credential}} is not
-                          a <span class="idlMemberType">DOMString</span>, then
-                          [= exception/throw =] an {{InvalidAccessError}}.
+                          If <var>parsedURL</var>'s' [=url/scheme=] is <code>"turn"</code> or
+                          <code>"turns"</code>, and neither
+                          <var>server</var>.{{RTCIceServer/username}} nor
+                          <var>server</var>.{{RTCIceServer/credential}}
+                          [=map/exist=], then [= exception/throw =] an {{InvalidAccessError}}.
```
Comment (alvestrand) [inline webrtc.html:3123]: I don't think it helps to have credential without username. The previous text insisted that both need to exist.

File: webrtc.html [hunk 2]
```diff
@@ -3073,44 +3073,54 @@ <h4>
...
+                        </ul>
+                      </li>
+                      <li>
+                        <p>If <var>parsedURL</var>'s [=url/scheme=] is not implemented by the
+                        user agent, then [=exception/throw=] a {{NotSupportedError}}.</p>
+                      </li>
+                      <li>
+                        <p>Let <var>hostAndPortURL</var> be result of
+                        <a data-cite="!url#concept-url-parser">parsing</a> the concatenation of
+                        `"https://"` and <var>parsedURL</var>'s [=url/path=].</p>
```
Comment (alvestrand) [inline webrtc.html:3100]: This is a decidedldy odd way of parsing the arguments to the URL.
It assumes that 7064 and 7065 are completely consistent with the relevant parts of the URL spec. I'd prefer to have the URL spec learn to parse TURN URLs - @annevk may have opinions on the proper way to do this.

**Review by alvestrand — CHANGES_REQUESTED** [review]
The username/password thing needs updating. As to the appropriateness of running the URL algorithm twice, once upon a synthetic URL, I'll let @annevk have an opinion on that.

File: webrtc.html (hunk 1 above)
Comment (jan-ivar) [reply webrtc.html:3123]: Whoops, I inverted the logic incorrectly. Thanks for spotting that.

**Review by alvestrand — APPROVED** [review]

**Pull request merged**